frontend/src/components/search/__tests__/AdvancedSearch.test.tsx

```diff
@@ -0,0 +1,213 @@
+        mockGetAllCategories.mockReturnValue(fetchedCategories)
+        const {getByTestId, getByRole, getByText} = renderPage(queryClient)
+        await waitFor(() => getByTestId("advancedSearchComponent"))
+        const selectField = getByRole("combobox")
+        const selectOption = (getByText("Wybierz kategorię") as HTMLOptionElement)
+        const inputField = getByRole("textbox")
+        const addRuleButton = getByRole("button", {
+            name: /dodaj regułę/i
+        })
+
+        await user.selectOptions(selectField, "Tytuł")
+        await user.type(inputField, "Przykładowy tytuł")
+        await user.click(addRuleButton)
+
+        expect(selectOption.selected).toBe(true)
+        expect(inputField).toHaveValue("")
+    })
+
+    test("rule deleting works correctly", async () => {
+        mockGetAllCategories.mockReturnValue(fetchedCategories)
+        const {getByTestId, getByRole} = renderPage(queryClient)
+        await waitFor(() => getByTestId("advancedSearchComponent"))
+        const selectField = getByRole("combobox")
+        const inputField = getByRole("textbox")
+        const addRuleButton = getByRole("button", {
+            name: /dodaj regułę/i
+        })
+
+        await user.selectOptions(selectField, "Tytuł")
+        await user.type(inputField, "Przykładowy tytuł")
+        await user.click(addRuleButton)
+
+        await user.selectOptions(selectField, "Rok")
+        await user.type(inputField, "1410")
+        await user.click(addRuleButton)
+
+        await user.click(getByTestId("delete-Tytuł"))
+
+        await user.selectOptions(selectField, "Artyści")
+        await user.type(inputField, "Jan Nowak")
+        await user.click(addRuleButton)
+
+        await user.selectOptions(selectField, "Tytuł.Podtytuł")
+        await user.type(inputField, "Przykładowy podtytuł")
+        await user.click(addRuleButton)
+
+        await user.click(getByTestId("delete-Tytuł.Podtytuł"))
+
+        expect(getByTestId("rules-container")).toMatchSnapshot()
+    })
+
+    test("error message shows up when user tries to add rule with category name already present in another rule", async () => {
+        mockGetAllCategories.mockReturnValue(fetchedCategories)
+        const {getByTestId, getByRole, getByText} = renderPage(queryClient)
+        await waitFor(() => getByTestId("advancedSearchComponent"))
+        const selectField = getByRole("combobox")
+        const inputField = getByRole("textbox")
+        const addRuleButton = getByRole("button", {
+            name: /dodaj regułę/i
+        })
+
+        await user.selectOptions(selectField, "Tytuł")
+        await user.type(inputField, "Przykładowy tytuł")
+        await user.click(addRuleButton)
+
+        await user.selectOptions(selectField, "Tytuł")
+        await user.type(inputField, "Przykładowy tytuł 2")
+        await user.click(addRuleButton)
+
+        expect(getByText(/reguła z tą nazwą kategorii już istnieje./i)).toMatchSnapshot()
```

Comment:
Tutaj jak sprawdzasz konkretny tekst to bardziej toBeInTheDocument, a toMatchSnapshot używasz w momencie jak chcesz sprawdzić jak się całość komponentu renderuje lub jego większy fragment.